code/game/objects/items/devices/boombox.dm

```diff
@@ -12,131 +12,96 @@
 	var/volume = 40
 	var/frequency = 1
 	var/datum/sound_token/sound_token
+	var/datum/track/current_track
 	var/list/datum/track/tracks
 	var/sound_id
 
-/obj/item/device/boombox/attack_self(var/mob/user)
-	interact(user)
+/obj/item/device/boombox/New()
+	..()
+	sound_id = "[/obj/item/device/boombox]_[sequential_id(/obj/item/device/boombox)]"
```

Comment:
why move this from init to new? code should be in initialize unless there's a need